Content.Shared/Humanoid/SkinColorationPrototype.cs

```diff
@@ -289,14 +299,34 @@ public Color ClosestSkinColor(Color color)
         var hsl = Color.ToHsl(color);
 
         if (Hue is (var minHue, var maxHue))
-            hsl.X = Math.Clamp(hsl.X, minHue, maxHue);
+            hsl.X = ClampHue(hsl.X, minHue, maxHue);
+        if (Saturation is (var minSat, var maxSat))
+            hsl.Y = Math.Clamp(hsl.Y, minSat, maxSat);
+        if (Lightness is (var minLight, var maxLight))
+            hsl.Z = Math.Clamp(hsl.Z, minLight, maxLight);
 
-        if (Saturation is (var minSaturation, var maxSaturation))
-            hsl.Y = Math.Clamp(hsl.Y, minSaturation, maxSaturation);
+        return Color.FromHsl(hsl);
+    }
 
-        if (Lightness is (var minValue, var maxValue))
-            hsl.Z = Math.Clamp(hsl.Z, minValue, maxValue);
+    private static bool IsHueInRange(float hue, float minHue, float maxHue)
+    {
+        if (minHue > maxHue) // Wraps around 1.0 (e.g., reds)
+            return hue >= minHue - Epsilon || hue <= maxHue + Epsilon;
+        return hue >= minHue - Epsilon && hue <= maxHue + Epsilon;
+    }
 
-        return Color.FromHsl(hsl);
+    private static float ClampHue(float hue, float minHue, float maxHue)
+    {
+        if (minHue > maxHue) // Wraps around 1.0
+        {
+            if (hue >= minHue || hue <= maxHue)
+                return hue;
+
+            var mid = (minHue + maxHue + 1) / 2f;
```

Comment:
```suggestion
            var mid = (minHue + maxHue) / 2f;
```
shouldn't the midpoint be this?